src/bin/help.ts

```diff
@@ -0,0 +1,182 @@
+import chalk from "chalk";
+
+import { logPipelessLine } from "../shared/cli/lines.js";
+import { allArgOptions } from "../shared/options/args.js";
+
+interface HelpTextSection {
+	sectionHeading: string;
+	subsections: {
+		flags: Flag[];
+		subheading: string | undefined;
+		warning: string | undefined;
+	}[];
+}
+
+interface Flag {
+	description: string;
+	flag: string;
+	type: string;
+}
+
+interface Option {
+	description: string;
+	docsSection: string;
+	type: string;
+}
+
+function logHelpTextSection(section: HelpTextSection): void {
+	logPipelessLine();
+
+	logPipelessLine(`   ${chalk.black.bgGreenBright(section.sectionHeading)}`);
+
+	for (const subsection of section.subsections) {
+		if (typeof subsection.warning === "string") {
+			logPipelessLine(chalk.yellow(subsection.warning));
+		}
+
+		if (typeof subsection.subheading === "string") {
+			logPipelessLine(chalk.green(subsection.subheading));
+		}
+
+		for (const option of subsection.flags) {
+			const { description, flag, type } = option;
+			logPipelessLine(
+				chalk.cyan(
+					`
+      --${flag}${
+				type !== "boolean" ? ` (${chalk.cyanBright(type)})` : ""
+			}: ${description}`,
+				),
+			);
+		}
+	}
+}
+
+function createHelpTextSections(options: object): HelpTextSection[] {
```

Comment:
> `options: object`

[Cleanup] Well now this why there are so many `as` assertions 😄 `object` is a super vague type. It just guarantees that the value is an object, not any particular properties on it. 

A more TypeScript-happy approach would be to...

1. Make an interface (`ArgOption`?) that includes `description`, `docsSection`, and `type`, then use it here.
2. Have the `docsSection` type be a union like `"core" | "opt-out" | ...`
3. Have `allArgsOptions` be declared as `satisfies Record<string, ArgOption>`

Reply:
I made some changes along these lines, thank you!